OrbitGl/SamplingReport.cpp

```diff
@@ -80,8 +80,8 @@ void SamplingReport::OnSelectAddress(uint64_t address, ThreadID thread_id) {
     if (selected_address_ != address || selected_thread_id_ != thread_id) {
       selected_address_ = address;
       selected_thread_id_ = thread_id;
-      UpdateDisplayedCallstack();
```

Comment:
I don't understand this, suppose that we are doing for the case that the user switch between tabs. But doesn't have more sense instead of updating this every time, to really update the thread_id when we move to another tab?

Reply:
I'm happy that you point out this issue. It also bothers me for a while.

So for example, if we have three thread tabs: tab 1, tab 2 and tab 3. Both tab 1 and 2 have selections but tab 3 don't.
If we click the thread tab following the order: tab 2 --> tab 3 --> tab 2, the UpdateDisplayedCallstack() will not be called if we don't move it out of the if loop.

The reason is that, when we switch to a new thread tab without making any selection, the selected_address_ and selected_thread_id_ are not updated, so if we switch back to the last tab which has selections, the selected_thread_id and  selected_address_ are still the same so the callstack widget will not be updated.

Reply:
Yes, I supposed that when I erased that line and playing a little with tabs.

But in my opinion doesn't have any sense being in a tab with an outdated reference to selected_thread_id_. This patch could work now, but could generate another issue later. So,

1- We can add at least one comment explaining the necessity of having this line out of that block.
2- Or better again, trying to update the thread_id when we move to another tab and not only when we select a row.

Reply:
Thanks for the suggestion. Now I keep both the selected function address and thread id to be updated: the selected_address is set as kInvalidFunctionAddress to indicate there is no selections; and also the thread id is always the current selected one.